### DISCUSSION TITLE
Not able to easily update the PR to the target the latest HEAD commit on the target branch

### DISCUSSION THREAD

**Discussion by dprotaso** [community/community#114511] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

It seems like re-running actions don't updated the merge commit of a PR.

For example:
- I open a PR against the main branch
- main is at revision 1
- there's bug in main which fails the actions in the PR mentioned above
- I push a fix to main - it's now a revision 2
- If I re-run the failed actions in the PR it still runs the actions against the same merge commit and fails

Is there a behaviour I'm missing as to why the merge commit doesn't update and re-running the actions should pass?

I'm aware you can close/reopen the PR but it seems like a weird UX issue. 

It would make sense for there to be a button or something on the PR page to re-run against latest target branch HEAD.

**Comment by ethanwater** [comment]
Ensure that your workflow is considering the correct branches you're pushing too. If not, it'll consider the bugged branch instead.

**Reply by dprotaso** [reply]
Sorry I don't understand what you mean. Can you clarify?

What I've been told is

> [Re-running a workflow](https://docs.github.com/en/actions/managing-workflow-runs/re-running-workflows-and-jobs#about-re-running-workflows-and-jobs) or jobs in a workflow uses the same GITHUB_SHA (commit SHA) and GITHUB_REF (Git ref) of the original event that triggered the workflow run.
>
> In this case, if you are using actions/checkout to fetch repository content - [it by default will fetch the SHA of the triggering event](https://github.com/actions/checkout/blob/main/action.yml#L7-L11). Using your example, it will fetch the repository at the state of the initial merge commit from the pull_request event that initially triggered the run.
